### ISSUE TITLE
Understanding For condition on BurnRate alerts

### ISSUE THREAD

**Issue by tbuchier** [metalmatze/slo-libsonnet#52]
Hello,

I'm wondering why the generated alerts set a `for` time ?

```
for: 2m
for: 15m
for: 1h
for: 3h
```

The source https://sre.google/workbook/alerting-on-slos/#5-multiple-burn-rate-alerts doesn't seems to be using them.

Why are they needed, and why those values (seems to be half of the range of the short window) ?
I though the 2 ranges combination was enough to avoid firing to soon.

Thanks !

**Comment by metalmatze** [comment]
Good question!
We could indeed live without them, most likely. 
@beorn7 wrote a fantastic blog post a while ago which touches on this too:

> The for clause is essentially a simple but effective defense against statistical outliers. For example, if a service has just started to receive traffic after a launch or an undrain operation, the error rate over any time window will be dominated by the very short time it has received traffic at all. Without the for clause, even a very short and modest initial error spike (maybe caused by cold caches) will trigger all the alerts.

Read the full blog post here: https://developers.soundcloud.com/blog/alerting-on-slos

tl;dr trade offs :man_shrugging:

**Comment by beorn7** [comment]
And both choices were deliberate. @metalmatze correctly quoted me on the reason why they are in the SoundCloud approach (and now here). I had the opportunity to review the source @tbuchier has quoted before it was published, and I suggested to add `for` clauses, but the authors took the deliberate decision to not use them to avoid the additional delay, added to the response time.

Another way to deal with statistical artifacts is a more advanced statistical approach, where you calculate confidence intervals based on the amount of traffic you have seen during the evaluation period, and then only alert if there is good statistical significance that a problem exist.

**Comment by tbuchier** [comment]
I see, thanks a lot for the quick responses ! 😃
